diff --git a/crates/vfio-ioctls/BUILD b/crates/vfio-ioctls/BUILD
new file mode 100644
index 000000000..f0f41ac2c
--- /dev/null
+++ crates/vfio-ioctls/BUILD
@@ -0,0 +1,21 @@
+load("@rules_rust//rust:defs.bzl", "rust_library")
+
+rust_library(
+    name = "vfio-ioctls",
+    srcs = glob(["src/*.rs"]),
+    crate_features = [
+        "kvm",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@crates//:byteorder",
+        "@crates//:kvm-bindings",
+        "@crates//:kvm-ioctls",
+        "@crates//:libc",
+        "@crates//:log",
+        "@crates//:thiserror",
+        "@crates//:vfio-bindings",
+        "@crates//:vm-memory",
+        "@crates//:vmm-sys-util",
+    ],
+)
+